src/Lecture3.hs

```diff
@@ -208,11 +228,30 @@ types that can have such an instance.
 -}
 
 -- instance Foldable Weekday where
+-- Impossible, Foldable needs kind * -> *, is of kind *
 -- instance Foldable Gold where
+-- Impossible, Foldable needs kind * -> *, is of kind *
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
-
+-- Impossible, Foldable needs kind * -> *, is of kind *
+instance Foldable List1 where
+  foldMap :: Monoid b => (a -> b) -> List1 a -> b
+  foldMap f (List1 a [])     = f a
+  foldMap f (List1 a [b])    = f a <> f b
+  foldMap f (List1 a (b:bs)) = f a <> foldMap f (List1 b bs)
```

Comment:
You don't need to implement a manual recursion here. You can reuse the `Foldable` instance for lists and write this in a more concise way